src/tools/rbd/action/PersistentCache.cc

```diff
@@ -0,0 +1,234 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "tools/rbd/ArgumentTypes.h"
+#include "tools/rbd/Shell.h"
+#include "tools/rbd/Utils.h"
+#include "include/types.h"
+#include "include/rbd_types.h"
+#include "include/stringify.h"
+#include "common/errno.h"
+#include "common/Formatter.h"
+#include <iostream>
+#include <boost/program_options.hpp>
+
+namespace rbd {
+namespace action {
+namespace persistent_cache {
+
+namespace at = argument_types;
+namespace po = boost::program_options;
+
+namespace {
+
+const std::string IMAGE_CACHE_STATE = ".librbd/image_cache_state";
+
+} // anonymous namespace
+
+static int do_persistent_cache_metadata_get(librbd::Image& image)
+{
+  int r;
+  uint64_t features;
+  r = image.features(&features);
+  if (r < 0) {
+    return r;
+  }
+
+  if (features & RBD_FEATURE_DIRTY_CACHE) {
+    std::string image_cache_str;
+    int r = image.metadata_get(IMAGE_CACHE_STATE, &image_cache_str);
+    if (r < 0) {
+      std::cerr << "rbd: failed to get image cache state: "
+                << cpp_strerror(r) << std::endl;
+    }
+    std::cout << image_cache_str << std::endl;
```

Comment:
Does this bring any value?  The exact same string is already output by `rbd status`.

Rather than adding a duplicate command I'd rather we try and address the main issue with image cache status -- the fact that the report is borderline useless and causes more harm than good because it is not updated while the image is open (https://tracker.ceph.com/issues/50614).

Reply:
sorry, I only recently found the confusion of status information. I added `TODO` to this PR and I will resolve it in the near future. thanks for reminding.